share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Your question 1: Ah, I see now where the mix-up is coming from. Vendors provide implementations of pthreads API spec. The spec does not prescribe what e.g. `pthread_t` should be mapped to. So vendors' implementations consist of both header files and the respective C implementations. Vendors define all types as they wish.

Your question 2: I honestly had no idea that kind of linking would work at all without throwing any errors or at least warnings. I have to say it looks a bit worrisome, but maybe it would work, not sure. At the very least it might confuse DSA if the types do not match and things will get collapsed. I have to think a bit more about this. In the meantime, Monty could also test it on an actual pthreads example.